### PR TITLE
KTOR-3667 Rewrite WebJars plugin to new API

### PR DESCRIPTION
**Pull request by Ololoshechkin**
https://youtrack.jetbrains.com/issue/KTOR-3667

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt [hunk 1]
```diff
@@ -1,126 +1,95 @@
...
     /**
-     * Plugin installation companion.
+     * Makes no effect. Will be dropped in future releases.
      */
-    public companion object Plugin : ApplicationPlugin<ApplicationCallPipeline, Configuration, Webjars> {
-
-        override val key: AttributeKey<Webjars> = AttributeKey("Webjars")
+    @Suppress("unused")
+    @Deprecated("This is no longer used and will be dropped in future releases.", level = DeprecationLevel.ERROR)
+    public var zone: ZoneId = ZoneId.systemDefault()
```
Comment (olme04) [inline ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt:41]: may be drop it now? as it's 2.0?

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt [hunk 2]
```diff
@@ -1,126 +1,95 @@
...
+}
 
-        override fun install(
-            pipeline: ApplicationCallPipeline,
-            configure: Configuration.() -> Unit
-        ): Webjars {
-            val configuration = Configuration().apply(configure)
+private fun extractWebJar(path: String, knownWebJars: Set<String>, locator: WebJarAssetLocator): String {
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt:44]: consider extracting public API to the separate file to make it easier to inspect

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt [hunk 3]
```diff
@@ -1,126 +1,95 @@
...
+            val fullPath = call.request.path()
+            if (fullPath.startsWith(webjarsPrefix) &&
+                call.request.httpMethod == HttpMethod.Get &&
+                fullPath.last() != '/'
+            ) {
+                val resourcePath = fullPath.removePrefix(webjarsPrefix)
+                try {
+                    val location = extractWebJar(resourcePath, knownWebJars, locator)
+                    val stream = WebjarsConfig::class.java.classLoader.getResourceAsStream(location) ?: return@onCall
+                    call.respond(
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt:79]: Could you add the test when the call is already handled by a different plugin before `Webjars`?

**Review by e5l — CHANGES_REQUESTED** [review]

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt (hunk 1 above)
Comment (Ololoshechkin) [reply ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt:41]: Will drop, thanks

**Review by rsinukov — APPROVED** [review]

**Review by e5l — APPROVED** [review]
Good job, lgtm

**Pull request merged**